ext/curl/multi.c

```diff
@@ -376,6 +377,10 @@ PHP_FUNCTION(curl_multi_info_read)
 			   being done in add_assoc_resource */
 			Z_ADDREF_P(pz_ch);
 
+			/* we must save result to be able to read error message */
+			ch = (php_curl*)zend_fetch_resource(Z_RES_P(pz_ch), le_curl_name, le_curl);
+			SAVE_CURLM_ERROR(ch, tmp_msg->data.result);
```

Comment:
Possibly write this out as `ch->err.no = tmp_msg->data.result`, as this is not a CURLM error. (It still works because the fields happen to be named the same on both structures).

Reply:
Maybe it's better to move `SAVE_CURL_ERROR` macro into `php_curl.h`, or not? It has been changed in 5d3337076b8dc8d4427a5c6259429972eab18db8 though, but only in 7.4.

Reply:
Exporting the macro sounds reasonable.